crates/rolldown/src/module_loader/module_task.rs

```diff
@@ -87,10 +88,24 @@ impl ModuleTask {
     let mut hook_side_effects = self.resolved_id.side_effects.take();
     let mut sourcemap_chain = vec![];
     let mut warnings = vec![];
+    let id = ModuleId::new(ArcStr::clone(&self.resolved_id.id));
 
     // Add watch files for watcher recover if build errors occurred.
     self.ctx.plugin_driver.watch_files.insert(self.resolved_id.id.clone());
 
+    self.ctx.plugin_driver.set_module_info(
```

Comment:
Considering this `set_module_info` we have `create module info` for each module three times in make phase.